src/App.tsx

```diff
@@ -9,39 +9,46 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { loadUsers } from './features/users/usersSlice';
+import { selectAuthor } from './features/author/authorSlice';
+import {
+  clearPosts,
+  loadPosts,
+  selectPosts,
+} from './features/posts/postsSlice';
+import {
+  selectSelectedPost,
+  setSelectedPost,
+} from './features/selectedPost/selectedPostSlice';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+  const { author } = useAppSelector(selectAuthor);
+  const { posts, loaded, hasError } = useAppSelector(selectPosts);
+  const { selectedPost } = useAppSelector(selectSelectedPost);
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const dispatch = useAppDispatch();
 
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
-
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  const loadUserPosts = useCallback(
+    (userId: number) => {
+      dispatch(loadPosts(userId));
+    },
+    [dispatch],
+  );
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(setSelectedPost(null));
 
     if (author) {
       loadUserPosts(author.id);
     } else {
-      setPosts([]);
+      dispatch(clearPosts());
     }
-  }, [author]);
+  }, [author, dispatch, loadUserPosts]);
```

Comment:
Ensure that the `loadUserPosts` function is correctly included in the dependency array of this `useEffect` hook. If `loadUserPosts` changes, the effect should re-run. This is important to avoid potential bugs related to stale closures.